addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/handler/NetworkHandler.java

```diff
@@ -50,20 +52,13 @@ public void handleCommand(ChannelUID channelUID, Command command) {
         if (command instanceof RefreshType) {
             switch (channelUID.getId()) {
                 case CHANNEL_ONLINE:
-                    try {
-                        State state = networkService.updateDeviceState() < 0 ? OnOffType.OFF : OnOffType.ON;
-                        updateState(CHANNEL_ONLINE, state);
-                    } catch (InvalidConfigurationException invalidConfigurationException) {
-                        updateStatus(ThingStatus.OFFLINE);
-                    }
+                    // The GUI requested a refresh. We send the cached value immediately and reset the refresh timer to
+                    // get the device reachable state as fast as possible.
```

Comment:
I didn't know about the ExpiringCache. Is there an API news feed existing somewhere?

Setting a new timeout for the scheduled job makes sense to me, yes. To my understanding the user expresses the wish that as soon as he knows about the (non-)reachable state of a device, he wants the next check to happen after the given period of time. So if the GUI causes a state refresh earlier in time, the next periodic check can be postponed.

I'll add logic for rate limiting / to prevent a ping while there is still one executed.